src/GeoBounds.cpp

```diff
@@ -1,40 +1,69 @@
-// Copyright © Matt Jones and Contributors. Licensed under the MIT License (MIT). See LICENCE.md in the repository root for more information.
+// Copyright © Matt Jones and Contributors. Licensed under the MIT License (MIT). See License.md in the repository root for more information.
 #include "GeoBounds.h"
 #include <math.h>
 
 namespace NovelRT {
-GeoBounds::GeoBounds(const NovelRT::GeoVector<float>& position, const NovelRT::GeoVector<float>& size, float rotation) : _position(position) {
-
-  _corners[0] = GeoVector<float>(position.getX() - (size.getX() / 2), position.getY() - (size.getY() / 2));
-  _corners[1] = GeoVector<float>(position.getX() + (size.getX() / 2), position.getY() - (size.getY() / 2));
-  _corners[2] = GeoVector<float>(position.getX() + (size.getX() / 2), position.getY() + (size.getY() / 2));
-  _corners[3] = GeoVector<float>(position.getX() - (size.getX() / 2), position.getY() + (size.getY() / 2));
-
-  rotateToAngle(rotation);
+GeoBounds::GeoBounds(const NovelRT::GeoVector<float>& position, const NovelRT::GeoVector<float>& size, float rotation) :
+    _position(position),
+    _rotation(rotation),
+    _size(size) {
 }
 
 bool GeoBounds::pointIsWithinBounds(const GeoVector<float>& point) const {
-  if (point.getX() >= _corners[0].getX() && point.getX() <= _corners[2].getX() && point.getY() <= _corners[2].getY()
-      && point.getY() >= _corners[0].getY())
-    return true;
 
   return false;
 }
+
+GeoVector<float> GeoBounds::getCornerInLocalSpace(int index) const {
+  GeoVector<float> returnValue;
+  auto size = getSize();
+  switch (index) {
+  case 0:returnValue = GeoVector<float>(-(size.getX() / 2), -(size.getY() / 2));
```

Comment:
Did you intend for the case expression to be on the same line as the case and with no space?

Reply:
Oof. No. Will fix.